### DISCUSSION TITLE
Lesson : 9 (One big test)

### DISCUSSION THREAD

**Discussion by today988** [Cyfrin/foundry-full-course-cu#1582] [Q&A]
When i run,

forge test --fork-url $SEPOLIA_RPC_URL

I got,

Running 1 test for test/unit/RaffleTest.t.sol:RaffleTest
[FAIL. Reason: setup failed: failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string] setUp() (gas: 0)
Traces:
  [19424858] RaffleTest::setUp()
    ├─ [19294109] → new DeployRaffle@0xFEfC6BAF87cF3684058D62Da40Ff3A795946Ab06
    │   └─ ← 96181 bytes of code
    ├─ [62488] DeployRaffle::run()
    │   ├─ [25224] → new <unknown>@0x6d2eed85750d316088343D6d5e91ca59eb052768
    │   │   ├─ [0] VM::envUint("PRIVATE_KEY") [staticcall]
    │   │   │   └─ ← failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string
    │   │   └─ ← 0 bytes of code
    │   └─ ← failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string
    └─ ← failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 2.94s

Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

**Comment by EngrPips** [comment]
Hello @today988 , please add `0x` to the front of your private key

**Reply by aminsammara** [reply]
@EngrPips thanks! Where can I understand about (the math?) re: why 0x is needed here? 

Edit: Found out that we typically use 0x to indicate hexadecimal from decimal numbers. I thought I heard Pat say the compiler / foundry is smart enough to realize that this is a hexadecimal string, which was the source of my confusion. 

No further action needed, thanks!

**Reply by EngrPips** [reply]
You are welcome, friend. I am not very sure what math you want to understand.
